File: 05-Proyecto/Backend/canchas/api/models/Reserva.js
/**
 * Reserva.js
 *
 * @description :: A model definition represents a database table/collection.
 * @docs        :: https://sailsjs.com/docs/concepts/models-and-orm/models
 */

module.exports = {

  attributes: {

    fecha: {
      type: 'string',
      required:true
    },
    horaInicial: {
      type: 'number',
      min: 0,
      required:true
    },

    horaFinal: {
      type: 'number',
      min: 0,
      required:true
    },
    
    codigoCli: {         // Nombre del fk para la relación
      model: 'cliente',   // Nombre del modelo a relacionar (padre) 
      required: true   // OPCIONAL-> Simpre se ingrese el fk
    },
   
     codigoCancha: {         // Nombre del fk para la relación
      model: 'cancha',   // Nombre del modelo a relacionar (padre) 
      required: true   // OPCIONAL-> Simpre se ingrese el fk
    },
   

  },

};

File: 05-Proyecto/ProyectoIndividual/app/src/main/java/com/example/proyectoindividual/modulo_reservas/Reserva.kt
package com.example.proyectoindividual.modulo_reservas

import com.example.proyectoindividual.clases_auxiliares.ClienteAux
import com.example.proyectoindividual.clases_auxiliares.CanchaAux

class Reserva(
    var id: Int?,
    var fecha: String?,
    var horaInicial: Int,
    var horaFinal: Int,
    var codigoCli: ClienteAux?,
    var codigoCancha: CanchaAux?
) {

}